src/server/services/wms/qgswmsgetcapabilities.cpp

```diff
@@ -703,6 +759,945 @@ namespace QgsWms
+    QString rootName = QgsServerProjectUtils::wmsRootName( *project );
+    if ( rootName.isEmpty() )
+    {
+      QDomText layerParentNameText = doc.createTextNode( project->title() );
+      layerParentNameElem.appendChild( layerParentNameText );
+    }
+    else
+    {
+      QDomText layerParentNameText = doc.createTextNode( rootName );
+      layerParentNameElem.appendChild( layerParentNameText );
+    }
+    layerParentElem.appendChild( layerParentNameElem );
+
+    // Root Layer title
+    QDomElement layerParentTitleElem = doc.createElement( QStringLiteral( "Title" ) );
+    QDomText layerParentTitleText = doc.createTextNode( project->title() );
+    layerParentTitleElem.appendChild( layerParentTitleText );
+    layerParentElem.appendChild( layerParentTitleElem );
+
+    // Root Layer tree name
+    if ( projectSettings )
+    {
+      QDomElement treeNameElem = doc.createElement( QStringLiteral( "TreeName" ) );
+      QDomText treeNameText = doc.createTextNode( project->title() );
+      treeNameElem.appendChild( treeNameText );
+      layerParentElem.appendChild( treeNameElem );
+    }
+
+    appendLayersFromTreeGroup( doc, layerParentElem, serverIface, project, version, request, projectLayerTreeRoot, projectSettings );
+
+    combineExtentAndCrsOfGroupChildren( doc, layerParentElem, project, true );
+
+    return layerParentElem;
+  }
+
+  namespace
+  {
+
+    void appendLayersFromTreeGroup( QDomDocument &doc,
+                                    QDomElement &parentLayer,
+                                    QgsServerInterface *serverIface,
+                                    const QgsProject *project,
+                                    const QString &version,
+                                    const QgsServerRequest &request,
+                                    const QgsLayerTreeGroup *layerTreeGroup,
+                                    bool projectSettings )
+    {
+      bool useLayerIds = QgsServerProjectUtils::wmsUseLayerIds( *project );
+      bool siaFormat = QgsServerProjectUtils::wmsInfoFormatSia2045( *project );
+      QStringList restrictedLayers = QgsServerProjectUtils::wmsRestrictedLayers( *project );
+
+      QList< QgsLayerTreeNode * > layerTreeGroupChildren = layerTreeGroup->children();
+      for ( int i = 0; i < layerTreeGroupChildren.size(); ++i )
+      {
+        QgsLayerTreeNode *treeNode = layerTreeGroupChildren.at( i );
+        QDomElement layerElem = doc.createElement( QStringLiteral( "Layer" ) );
+        if ( treeNode->nodeType() == QgsLayerTreeNode::NodeGroup )
+        {
+          QgsLayerTreeGroup *treeGroupChild = static_cast<QgsLayerTreeGroup *>( treeNode );
+
+          QString name = treeGroupChild->name();
+          if ( restrictedLayers.contains( name ) ) //unpublished group
+          {
+            continue;
+          }
+
+          if ( projectSettings )
+          {
+            layerElem.setAttribute( QStringLiteral( "visible" ), treeGroupChild->isVisible() );
```

Comment:
As I understand (and actually observe) this sets the `visible` attribute on "group" layers only. In previous versions of QGIS Server (2.18 for example) the `visible` attribute was set on both "group" and "leaf" layers. Is this behaviour change intentional? Or is this a regression?

Ping @rldhont.

Reply:
It's more a regression than an intentional change.

Reply:
That's what I thought. Thanks for confirming, we'll work on a fix.